internal/notify/notifications.lua

```diff
@@ -159,6 +159,8 @@ local function for_wildlife_adv(fn, reverse)
             not unit.flags1.caged and
             not unit.flags1.chained and
             not dfhack.units.isHidden(unit) and
+            unit.relationship_ids.GroupLeader ~= dfhack.world.getAdventurer().id and
```

Comment:
you can factor out the `dfhack.world.getAdventurer().id` by putting a
`local adv_id = dfhack.world.getAdventurer().id`
above the `for_iter` call.
notifications take a significant amount of processing time, so we need to be careful about inefficiencies